halo2_frontend/src/plonk/circuit/constraint_system.rs

```diff
@@ -404,6 +410,14 @@ impl<F: Field> ConstraintSystem<F> {
                 if table.contains_simple_selector() {
                     panic!("expression containing simple selector supplied to lookup argument");
                 }
+                if !table.contains_fixed_col_or_selector() {
+                    panic!(
+                        "table expression supplied to lookup_any argument must include fixed column or selector"
+                    );
+                }
+                if table.degree() == 1 {
+                    panic!("table expression containing only fixed column supplied to lookup_any argument, should use `lookup` api instead of `lookup_any`");
+                }
```

Comment:
Note that this is checking a condition for **any** table expressions.  But actually we want to check that **all** table expressions are a fixed/selector column query.

Reply:
Hmm, I think the current checks on `lookup_any` api are for **all** table expressions. Am I wrong?
```
 pub fn lookup_any<S: AsRef<str>>(
        &mut self,
        name: S,
        table_map: impl FnOnce(&mut VirtualCells<'_, F>) -> Vec<(Expression<F>, Expression<F>)>,
    ) -> usize {
        let mut cells = VirtualCells::new(self);
        let table_map = table_map(&mut cells)
            .into_iter()
            .map(|(mut input, mut table)| {
                if input.contains_simple_selector() {
                    panic!("expression containing simple selector supplied to lookup argument");
                }
                if table.contains_simple_selector() {
                    panic!("expression containing simple selector supplied to lookup argument");
                }
                if !table.contains_fixed_col_or_selector() {
                    panic!(
                        "table expression supplied to lookup_any argument must include fixed column or selector"
                    );
                }
                if table.degree() == 1 {
                    panic!("table expression containing only fixed column supplied to lookup_any argument, should use `lookup` api instead of `lookup_any`");
                }
                input.query_cells(&mut cells);
                table.query_cells(&mut cells);
                (input, table)
            })
            .collect();
      let index = self.lookups.len();

        self.lookups
            .push(lookup::Argument::new(name.as_ref(), table_map));

        index
    }
```
It iterates every `input` and `table` expressions, and do the check.
So, if any of `table` expression do not match the check, it panics.